app/controllers/merchants/orders_controller.rb

```diff
@@ -0,0 +1,8 @@
+class Merchants::OrdersController <ApplicationController
+  def show
+    @merchant = current_user
+    @order = Order.find(params[:id])
+    @user = @order.user
+    @order_items = @order.order_items.map {|order_item| order_item if order_item.item.user_id == @merchant.id}.compact
```

Comment:
This should probably change.

Reply:
It works.

Reply:
it's grate.

Reply:
I'm not sure if this should be done in the model or not, but it certainly works.